src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -100,6 +100,10 @@ public void put(Snapshot.Key key, Delete delete) {
     deleteSet.put(key, delete);
   }
 
+  public boolean containsKey(Snapshot.Key key) {
+    return writeSet.containsKey(key) || readSet.containsKey(key);
```

Comment:
Should we check that the key is not in `deleteSet`?

This may be off-topic to this PR, but what happens if we get/scan after delete?

- `get(key1)` → `delete(key1)` → `get(key1)`
- `scan(key1...key10)` → `delete(key5)` → `scan(key1...key10)`

Note that `get` after `write` results in `CrudRuntimeException("reading already written data is not allowed")` raised from `Snapshot#get` while `get` after `delete` doesn't seem to be checked.